### PR TITLE
Add Gmail.User.attachment/3

### PR DESCRIPTION
**Pull request by killtheliterate**
...Still in progress, just thought I'd open a WIP PR.

### CODE REVIEW

**Comment by killtheliterate** [comment]
Once I have this working (fingers crostttttt), I'll squash everything down.

**Comment by killtheliterate** [comment]
I know it's a bit weird to open an in-progress PR: https://medium.com/practical-blend/pull-request-first-f6bb667a9b6#.ywltavjxm

**Comment by craigp** [comment]
I actually think I like the idea of a WIP PR

File: lib/gmail/message_attachment.ex [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  import Gmail.Base
+
+  @doc """
+  Gmail API documentation: https://developers.google.com/gmail/api/v1/reference/users/messages/attachments
+  """
+  defstruct attachmentId: "",
+    size: "",
+    data: ""
+
+  @type t :: %__MODULE__{}
```
Comment (talon) [inline lib/gmail/message_attachment.ex:18]: what is this part?

File: lib/gmail/message_attachment.ex [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  """
+  def handle_attachment_response(response) do
+    response
+    |> handle_error
+    |> case do
+      {:error, message} ->
+        {:error, message}
+      {:ok, raw_message} ->
+        {:ok, MessageAttachment.convert(raw_message)}
+      # :ok -> :ok # uh... i'm not sure why i should handle this case.
```
Comment (killtheliterate) [inline lib/gmail/message_attachment.ex:54]: @craigp while I was wiring up tests, I hit this case when using empty strings for message_id and attachment_id. Should I handle the case? I believe it's from `do_parse_response` ln:66 of http.ex

File: lib/gmail/message_attachment.ex (hunk 1 above)
Comment (killtheliterate) [inline lib/gmail/message_attachment.ex:18]: @talon the struct? It's the shape of this module's data. Or are you asking about `@type`. I was also wondering about that.

**Comment by killtheliterate** [comment]
@craigp I think this is ready for merge.

File: lib/gmail/message_attachment.ex (hunk 1 above)
Comment (talon) [inline lib/gmail/message_attachment.ex:18]: yeah the `@type` thing

File: lib/gmail/message_attachment.ex [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  def get(user_id, message_id, id) do
+    path = querify_params("users/#{user_id}/messages/#{message_id}/attachments/#{id}", [], %{})
+
+    {:get, base_url, path}
+  end
+
+  @doc """
+  Converts a Gmail API attachment resource into a local struct.
+  """
+  @spec convert(map) :: MessageAttachment.t
```
Comment (talon) [inline lib/gmail/message_attachment.ex:35]: oh I guess this is what `@type` is defining `MessageAttachement.t` which just boils down to the defstruct I think? Elixir is so dope!

Comment (killtheliterate) [inline lib/gmail/message_attachment.ex:35]: @talon yeah, I think it's `dialyzer`. Looking into it now 😹 - learning in public

Comment (talon) [inline lib/gmail/message_attachment.ex:35]: same. I didn't know how to search it so I just gazed until I had a reasonable guess

Comment (killtheliterate) [inline lib/gmail/message_attachment.ex:35]: @talon hmm, yea, abort on the `dialyzer` notion. [typespecs](http://elixir-lang.org/docs/stable/elixir/typespecs.html).

**Comment by craigp** [comment]
@talon @killtheliterate yep, it's just the typespec for this module. I'll review this sometime today guys, thanks for the contribution.

**Pull request merged**